### PR TITLE
Rdcc 1873 1

### PR DESCRIPTION
**Pull request by KOTLAPRA23**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDCC-1873

### Change description ###

feat:email value picking from header instead of query param

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/professionalapi/controller/external/OrganisationExternalControllerTest.java [hunk 1]
```diff
@@ -186,14 +191,15 @@ public void test_RetrieveOrganisationByIdentifier() {
     @Test
     public void test_RetrievePaymentAccountByUserEmail() {
         final HttpStatus expectedHttpStatus = HttpStatus.OK;
+        RequestContextHolder.setRequestAttributes(new ServletRequestAttributes(httpRequest));
 
         List<String> authorities = new ArrayList<>();
         authorities.add(TestConstants.PUI_USER_MANAGER);
         String email = "test@email.com";
+        when(httpRequest.getHeader(anyString())).thenReturn(email);
```
Comment (AdilOoze) [inline src/test/java/uk/gov/hmcts/reform/professionalapi/controller/external/OrganisationExternalControllerTest.java:199]: please add `verify()` for this `when().thenReturn()`

File: src/test/java/uk/gov/hmcts/reform/professionalapi/controller/external/OrganisationExternalControllerTest.java [hunk 2]
```diff
@@ -202,6 +208,27 @@ public void test_RetrievePaymentAccountByUserEmail() {
...
+    public void testRetrievePaymentAccountByUserEmailFromHeader() {
+        final HttpStatus expectedHttpStatus = HttpStatus.OK;
+
+        RequestContextHolder.setRequestAttributes(new ServletRequestAttributes(httpRequest));
+        List<String> authorities = new ArrayList<>();
+        authorities.add(TestConstants.PUI_USER_MANAGER);
+        String email = "test@email.com";
+        when(httpRequest.getHeader(anyString())).thenReturn(email);
+        when(jwtGrantedAuthoritiesConverterMock.getUserInfo()).thenReturn(userInfoMock);
+        when(userInfoMock.getRoles()).thenReturn(authorities);
```
Comment (AdilOoze) [inline src/test/java/uk/gov/hmcts/reform/professionalapi/controller/external/OrganisationExternalControllerTest.java:221]: please add `verify()` for these `when().thenReturn()`

File: src/test/java/uk/gov/hmcts/reform/professionalapi/controller/external/ProfessionalExternalUserControllerTest.java [hunk 3]
```diff
@@ -294,24 +300,68 @@ public void test_FindUserStatusByEmail() throws JsonProcessingException {
         newUserResponse.setIdamStatus("ACTIVE");
         ObjectMapper mapper = new ObjectMapper();
         String body = mapper.writeValueAsString(newUserResponse);
+        String email = "test@email.com";
+        RequestContextHolder.setRequestAttributes(new ServletRequestAttributes(httpRequest));
+        when(httpRequest.getHeader(anyString())).thenReturn(email);
+        ResponseEntity<NewUserResponse> responseEntity1 = new ResponseEntity<NewUserResponse>(newUserResponse,
+                HttpStatus.OK);
+
+        when(userProfileFeignClient.getUserProfileByEmail(anyString())).thenReturn(Response.builder()
```
Comment (AdilOoze) [inline src/test/java/uk/gov/hmcts/reform/professionalapi/controller/external/ProfessionalExternalUserControllerTest.java:309]: please add `verify()` for this `when().thenReturn()`

Comment (AdilOoze) [inline src/test/java/uk/gov/hmcts/reform/professionalapi/controller/external/ProfessionalExternalUserControllerTest.java:309]: please add `verify()` for these two `when().thenReturn()`

File: src/test/java/uk/gov/hmcts/reform/professionalapi/controller/external/OrganisationExternalControllerTest.java (hunk 2 above)
Comment (KOTLAPRA23) [reply src/test/java/uk/gov/hmcts/reform/professionalapi/controller/external/OrganisationExternalControllerTest.java:221]: ok.done

File: src/test/java/uk/gov/hmcts/reform/professionalapi/controller/external/ProfessionalExternalUserControllerTest.java (hunk 3 above)
Comment (KOTLAPRA23) [reply src/test/java/uk/gov/hmcts/reform/professionalapi/controller/external/ProfessionalExternalUserControllerTest.java:309]: done

File: src/test/java/uk/gov/hmcts/reform/professionalapi/controller/external/OrganisationExternalControllerTest.java (hunk 1 above)
Comment (KOTLAPRA23) [reply src/test/java/uk/gov/hmcts/reform/professionalapi/controller/external/OrganisationExternalControllerTest.java:199]: done

**Review by AdilOoze — APPROVED** [review]
Requested changes have been addressed 👍

File: src/integrationTest/java/uk/gov/hmcts/reform/professionalapi/util/ProfessionalReferenceDataClient.java [hunk 4]
```diff
@@ -67,8 +68,15 @@ public ProfessionalReferenceDataClient(int port, String issuer, Long tokenExpira
...
 
-    public Map<String, Object> findPaymentAccountsByEmailForExternalUsers(String email, String role) {
-        return getRequest("/refdata/external/v1/organisations" + "/pbas?email={email}", role, email);
+    public Map<String, Object> findPaymentAccountsByEmailFromHeader(String email, String role) {
+        return getRequestToGetEmailFromHeader("/refdata/internal/v1/organisations" + "/pbas?email={email}",
+                role, "", email);
+    }
+
+    public Map<String, Object> findPaymentAccountsByEmailFromHeaderForExternalUsers(String email, String role,
+                                                                                    String userId) {
```
Comment (shashank-rastogi) [inline src/integrationTest/java/uk/gov/hmcts/reform/professionalapi/util/ProfessionalReferenceDataClient.java:77]: should not you have two methods here. -
1. one which takes parameter from header (with no email id as query param)
2. another method which takes the email id parameter from query and having a check of param from header
and there should be subsequent test cases using each of the above?

Right now it tests only the scenario when both conditions - from header and query param are checked  and the final state where  only the header param would exist is not tested at all

**Review by shashank-rastogi — APPROVED** [review]

File: src/integrationTest/java/uk/gov/hmcts/reform/professionalapi/util/ProfessionalReferenceDataClient.java (hunk 4 above)
Comment (KOTLAPRA23) [reply src/integrationTest/java/uk/gov/hmcts/reform/professionalapi/util/ProfessionalReferenceDataClient.java:77]: as discussed closing the conversation

**Review by pksaxena01 — APPROVED** [review]
Approved

**Review by shreedhar-hmcts — APPROVED** [review]

**Review by mallikarjunputtana — APPROVED** [review]

**Pull request merged**